README.md

```diff
@@ -145,6 +153,165 @@ readData(dataInput) {
 }
 ```
 
+# Using Node.js Client with Hazelcast IMDG
+
+## 4. Distributed Events
+
+This chapter introduces Distributed Events which are Cluster Events and Distibuted Data Structure Events.
+
+### 4.1. Cluster Events
+
+You can add event listeners to a Hazelcast Node.js client. You can configure the following listeners to listen to the events on the client side.
+
+`Membership Listener`: Notifies when a member joins to/leaves the cluster, or when an attribute is changed in a member.
+
+`Distributed Object Listener`: Notifies when a distributed object is created or destroyed throughout the cluster.
+
+`Lifecycle Listener`: Notifies when the client is starting, started, shutting down, and shutdown.
+
+#### 4.1.1. Listening for Member Events
+
+You can add the following 3 different types of member events to the `ClusterService`.
+
+- `memberAdded`: A new member is added to the cluster.
+- `memberRemoved`: An existing member leaves the cluster.
+- `memberAttributeChanged`: An attribute of a member is changed. Please refer to [Defining Member Attributes](https://docs.hazelcast.org/docs/3.10.5/manual/html-single/index.html#defining-member-attributes) section in the Hazelcast IMDG Reference Manual to learn about member attributes.
+
+The `ClusterService` object exposes an `ClusterService.on()` function that allows one or more functions to be attached to member events emitted by the object.
+
+The following is a membership listener registration by using `ClusterService.on()` function.
+
+```javascript
+client.clusterService.on('memberAdded', function (member) {
+    console.log('Member Added: The address is', member.address.toString());
+});
+```
+
+The `memberAttributeChanged` has its own type of event named `MemberAttributeEvent`. When there is an attribute change on the member, this event is fired.
+
+Let’s take a look at the following example.
+
+```javascript
+client.clusterService.on('memberAttributeChanged', function (memberAttributeEvent) {
+    console.log('Member Attribute Changed: The address is', memberAttributeEvent.member.address.toString());
+});
+```
+
+####  4.1.2. Listening for Distributed Object Events
+
+The Distributed Object Listener events are invoked when a distributed object is created and destroyed throughout the cluster. After the events, a listener callback function is called. The type of the callback function should be `DistributedObjectListener`. The parameter of the function is `DistributedObjectEvent` including following fields:
```

Comment:
The events for distributed objects are invoked when they are created and destroyed in the cluster. After the events, ...